browser/brave_stats/brave_stats_updater.h

```diff
@@ -103,10 +105,6 @@ class BraveStatsUpdater : public ProfileManagerObserver {
   DISALLOW_COPY_AND_ASSIGN(BraveStatsUpdater);
 };
 
-// Creates the BraveStatsUpdater
-std::unique_ptr<BraveStatsUpdater> BraveStatsUpdaterFactory(
-    PrefService* pref_service);
-
```

Comment:
Now that this `BraveStatsUpdaterFactory()` method is gone, perhaps it would be worth having a comment over `class BraveStatsUpdater` clarifying that instances of this class will be **owned** by `BraveProcessImpl`, and therefore discouraging from manually instantiating more instances from other places (e.g. tests), as the existence of the `BraveStatsUpdaterFactory()` seemed to suggest.

Maybe also worth clarifying in that comment that class is meant to be instantiating during the early initialization stage of the browser process (and crucially before the primary profile has been loaded).

WDYT?

Reply:
Well, it would be totally reasonable and necessary to instantiate them for unit tests

Reply:
Oops, I guess what I was thinking of only applies to `BraveReferralsService` then?